test-crates/numpy-user/Cargo.toml

```diff
@@ -0,0 +1,13 @@
+[package]
+name = "numpy-user"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+name = "rust_ext"
+crate-type = ["cdylib"]
+
+[dependencies]
+pyo3 = { version = "0.17", features = ["extension-module"] }
+numpy = { version = "0.17", git = "https://github.com/PyO3/rust-numpy", branch = "python-requires-in-cargo-metadata" }
```

Comment:
I was wondering why NumPy 1.19 while our README says 1.16? (The choice predates my involvement, so I cannot say one is correct and the other is not. But I would like to understand if our README would need to be changed.)

I am also somewhat sceptical w.r.t. the upper bound as the situation with Python dependencies is significantly different from Rust dependencies due to the need for venv-wide unification, c.f. https://iscinumpy.dev/post/bound-version-constraints/